core/geometry/src/curve/CurveFactory.ts

```diff
@@ -426,101 +432,154 @@ export class CurveFactory {
     }
     return arcs;
   }
+
+  /** For a smooth curve, stroke and return unnormalized start/end tangents */
+  private static strokeSmoothCurve(
+    curve: IndexedXYZCollection | Point3d[] | CurvePrimitive | CurveChain,
+    options?: StrokeOptions,
+  ): SmoothCurveData | undefined {
+    let startTangent, endTangent: Vector3d | undefined;
+    if (curve instanceof CurvePrimitive) {
+      startTangent = curve.fractionToPointAndDerivative(0.0).direction;
+      endTangent = curve.fractionToPointAndDerivative(1.0).direction;
+      const strokes = LineString3d.create();
+      curve.emitStrokes(strokes, options);
+      curve = strokes.packedPoints;
+    } else if (curve instanceof CurveChain) {
+      startTangent = curve.startPointAndDerivative()?.direction;
+      endTangent = curve.endPointAndDerivative()?.direction;
+      const strokes = curve.getPackedStrokes(options);
+      if (!strokes)
+        return undefined;
+      curve = strokes;
+    } else if (Array.isArray(curve))
+      curve = new Point3dArrayCarrier(curve);
+    return {strokes: curve, startTangent, endTangent};
+  }
+
+  /**
+   * Align two bisector plane normals to given smooth rail tangents or optional overrides.
+   * * Optionally average the normals for physically closed rail.
+   */
+  private static alignFirstAndLastBisectorPlanes(
+    firstPlane: Plane3dByOriginAndUnitNormal,
+    lastPlane: Plane3dByOriginAndUnitNormal,
+    smoothRailData?: SmoothCurveData,
+    options?: MiteredSweepOptions,
+  ) {
+    const normal0 = options?.startTangent ?? (smoothRailData?.startTangent ?? firstPlane.getNormalRef());
```

Comment:
This is confusing to me. 

1- If `options.startTangent` is defined we assign it to `normal0`
2- Otherwise, if `smoothRailData.startTangent`  is defined we assign it to `normal0`
3- Otherwise, we assign `firstPlane.getNormalRef()` to `normal0`
so here we can set one of the 3.

In the original code, if `options.startTangent` was defined we assigned it to `normal0`. Otherwise, `firstPlane.getNormalRef()` 

Now my confusing is this new possible normal `smoothRailData.startTangent`. Why does it take precedence over `firstPlane.getNormalRef()` ?

Reply:
Now that we have expanded the rail type to allow smooth curves, we have another way to set the normals of the first/last planes. If the rail is smooth, we can compute its first/last tangent and set the first/last plane normals accordingly, and this will yield a better-looking mesh (as you initially observed in your earliest tests of `PolyfaceBuilder.addMiteredPipes`). Thus, rail start/end tangents are preferred if the rail is smooth. But tangents provided by the caller in the options are preferred over smooth rail tangents. The order of decreasing preference for setting the first/last plane normals is as follows:
1. tangent from the options (if given)
2. tangent from the rail (if it's smooth)
3. tangent from the rail polyline (computed by `PolylineOps.createBisectorPlanesForDistinctPoints`)